posthog/hogql/grammar/README.md

```diff
@@ -0,0 +1,32 @@
+## How to generate source code files from grammar
+
+Grammar is located inside `HogQLLexer.g4` and `HogQLParser.g4` files.
+
+To generate source code you need to install locally the `antlr4` binary:
+
+```bash
+brew install antlr
+```
+
+Then either run
+
+```bash
+pnpm run grammar:build
+```
+
+Or mess around with:
+
+```bash
+cd posthog/hogql/grammar
+antlr4 -Dlanguage=Python3 HogQLLexer.g4
+antlr4 -visitor -Dlanguage=Python3 HogQLParser.g4
+```
+
+Original ClickHouse ANTLR grammar from: https://github.com/ClickHouse/ClickHouse/blob/master/utils/antlr/ClickHouseParser.g4
+
+Changes with ClickHouse's grammar:
+- removed all statements except for "select"
+- support aliases with a string literal "as '🍄'"
```

Comment:
This was added to support column titles in data tables, but I'll likely change it again when working on symbol resolution (aliasing, namescapes, table identifiers ala `selecvt e.properties from events e`) later. It's probably better to keep comments for column titles.